test/Microsoft.AspNetCore.Antiforgery.Test/Internal/DefaultAntiforgeryTokenStoreTest.cs

```diff
@@ -236,9 +236,16 @@ public void GetCookieToken_CookieIsNotEmpty_ReturnsToken()
         }
 
         [Theory]
-        [InlineData(CookieSecurePolicy.Always, true)]
-        [InlineData(CookieSecurePolicy.None, null)]
-        public void SaveCookieToken(CookieSecurePolicy policy, bool? expectedCookieSecureFlag)
+        [InlineData(false, CookieSecurePolicy.SameAsRequest, null)]
+        [InlineData(true, CookieSecurePolicy.SameAsRequest, true)]
+        [InlineData(false, CookieSecurePolicy.Always, true)]
+        [InlineData(true, CookieSecurePolicy.Always, true)]
+        [InlineData(false, CookieSecurePolicy.None, null)]
+        [InlineData(true, CookieSecurePolicy.None, null)]
+        public void SaveCookieToken_HonorsCookieSecurePolicy_OnOptions(
```

Comment:
Could we split these out in to meaningful scenarios:

`SaveCookieToken_UsesRequestPolicy_WhenCookieSecurePolicyIsSameAsRequest`
`SaveCookieToken_CreatesSecurePolicyCookies_WhenCookieSecurePolicyIsAlways`
`SaveCookieToken_UsesValueFromConfig_WhenCookieSecurePolicyIsNone`

We can re-use the actual code as a private method but it'll be easier to understand what specific scenario is broken if this regresses.

Reply:
Hmm, I think it looks fine as of now. It isn't difficult to follow even if it regresses.